### PR TITLE
1558 review page and tabs

### PR DESCRIPTION
**Pull request by guatemartin**
Have tried to use Semantic as much as possible to do the heavy lifting.  Hopefully I haven't disturbed the desktop portion too much by using Grid columns in some places and leveraging the Stackable property of a grid.

1. Removed ProgressBar for mobile.  Fits well in simple 2 step processes, but in the more complicated it goes awry.
2. Removed Icon in the Review->Overview Tab.   Didn't seem necessary to provide in mobile
3. Tab headers wrap.  Thought about drop down as it might be cleaner looking, but tabs are easier to navigate.
4. Checked assignment tab looks ok with multiple stages.

### CODE REVIEW

File: semantic/src/site/globals/site.overrides
Comment (guatemartin) [inline semantic/src/site/globals/site.overrides:910]: Removed this min-width site wide.  Couldn't see it being of any benefit anywhere, and it would stuff up the mobile rendering.

File: semantic/src/site/globals/site.overrides
Comment (guatemartin) [inline semantic/src/site/globals/site.overrides:935]: Why 0.57?  That's just what seemed to look best.

File: src/containers/Review/assignment/ReviewSectionRow.tsx
Comment (guatemartin) [inline src/containers/Review/assignment/ReviewSectionRow.tsx:103]: I thought I would try to use Semantic's Grid Stackable property if possible.  If it's not what we want, I can find another way.

File: semantic/src/site/collections/menu.overrides
Comment (CarlosNZ) [inline semantic/src/site/collections/menu.overrides:10]: Good to be able to wrap the tabs. However, the upper ones look a little weird just "floating" above the rest of the row and not attached to the container beneath.

Not sure what the solution is. It's a slight improvement if you add the border to the bottom (it only goes on left, right, top normally), but still doesn't look great. Would it be reasonable to *switch* the order so that the "active" one was always on the bottom row, or would that be annoying?

Any other suggestions?

File: semantic/src/site/globals/site.overrides
Comment (CarlosNZ) [reply semantic/src/site/globals/site.overrides:936]: Screenshot 2023-11-07 at 12 05 40 PM

I reckon we can lose the "STAGE" label, cos it's clear from the coloured bubble that it's referring to stage, which will give more room for the other stuff.

And maybe, let it wrap so the "Stage" bubble can go to next line? (It'd be good if the text/arrow inside the drop-down can stay on a single line)

File: src/containers/Review/assignment/ReviewSectionRow.tsx
Comment (CarlosNZ) [reply src/containers/Review/assignment/ReviewSectionRow.tsx:103]: Screenshot 2023-11-07 at 12 12 29 PM
Yeah that's great, thanks. Only thing I'd suggest is for the rows within each block to be a bit closer together.

File: src/containers/Review/overview/Overview.tsx
Comment (CarlosNZ) [inline src/containers/Review/overview/Overview.tsx:41]: Screenshot 2023-11-07 at 12 16 18 PM

Just need to get the "Extend deadline" UI to wrap on mobile. (Hit me up in Telegram if you can't find an application with this set -- or just change the condition to `true` temporarily to make it show up.)

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Cool, looking much better. Good call to use the Grid properly to stack nicely.

See comments for some small adjustments.

Also, the Notes and Documents tabs could do with a little tidy up -- you want to make a new issue for them?
 
Screenshot 2023-11-07 at 12 21 12 PM
- Left justify sort/files UI
- Input box full width

Screenshot 2023-11-07 at 12 21 54 PM- 
- icon a bit smaller
- reduce spacing

File: src/containers/Review/ReviewWrapper.tsx
Comment (CarlosNZ) [inline src/containers/Review/ReviewWrapper.tsx:110]: Oh one more thing. The progress bar actually looks okay on mobile up to 3 items. Maybe we should keep it unless there are more that two stages (which makes 3 dots in the bar)?

Screenshot 2023-11-07 at 12 28 33 PM

(But just reduce the top/bottom space around it)

Comment (CarlosNZ) [reply src/containers/Review/ReviewWrapper.tsx:110]: Actually, don't worry about it. It's still a bit cut off. More trouble than it's worth I think.

**Comment by CarlosNZ** [comment]
Oh, couple of other things I just noticed after posting my review:

There is a min-width of 300px on the `#overview .left-item` class that is causing problems when we get narrow:
Screenshot 2023-11-07 at 12 37 09 PM
It's fine if you turn this off for mobile:

Screenshot 2023-11-07 at 12 37 15 PM


And can we reduce the padding around the Activity region to use more of the screen width (maybe drop the indents down a little too, if it helps?)

Screenshot 2023-11-07 at 12 37 28 PM

I think we can also revert the font-size (0.57 rem) on the Stage bubbles back to what it was, as with the other tweaks I suggested it's probably not necessary.

File: semantic/src/site/globals/site.overrides
Comment (guatemartin) [reply semantic/src/site/globals/site.overrides:935]: Text Label gone.  Left as much spacing btn dropdown and Stage Label as I could.  It's not much really.
assignment_tab

File: semantic/src/site/globals/site.overrides
Comment (CarlosNZ) [inline semantic/src/site/globals/site.overrides:2812]: Can you just add some left padding to make this row flush with the stuff beneath it?
Screenshot 2023-11-07 at 4 59 25 PM

File: src/containers/Review/assignment/ReviewSectionRow.tsx
Comment (guatemartin) [reply src/containers/Review/assignment/ReviewSectionRow.tsx]: Reduced spacing... we'll see if it's enough, or needs more...

File: semantic/src/site/collections/message.overrides
Comment (guatemartin) [reply semantic/src/site/collections/message.overrides:44]: With the removal of other properties that were limiting the space... this is now 1em, and it isn't causing issues.

File: src/containers/Review/overview/Overview.tsx
Comment (guatemartin) [reply src/containers/Review/overview/Overview.tsx:41]: Ohh, forgot this one...  Will get it

File: semantic/src/site/globals/site.overrides
Comment (CarlosNZ) [inline semantic/src/site/globals/site.overrides:2822]: Screenshot 2023-11-07 at 5 11 03 PM

I think we need a bit more tidying up of this layout. But lets leave that to another issue.

**Comment by guatemartin** [comment]
> Oh, couple of other things I just noticed after posting my review:
> 

I hadn't seen this one!  Thanks.  It was causing me grief and I couldn't find it!!

> There is a min-width of 300px on the `#overview .left-item` class that is causing problems when we get narrow: Screenshot 2023-11-07 at 12 37 09 PM It's fine if you turn this off for mobile:
> 
> Screenshot 2023-11-07 at 12 37 15 PM
> 

It was actually the margins that were huge.  Have reduced for mobile.

> And can we reduce the padding around the Activity region to use more of the screen width (maybe drop the indents down a little too, if it helps?)
> 
> Screenshot 2023-11-07 at 12 37 28 PM
>

There was a little more space, but I still left it at 0.7rem, otherwise it uses up a lot of space.... I left it at .7 thinking it looked good, but if not, 1rem it is.
 
> I think we can also revert the font-size (0.57 rem) on the Stage bubbles back to what it was, as with the other tweaks I suggested it's probably not necessary.

File: src/containers/Review/ReviewWrapper.tsx
Comment (CarlosNZ) [inline src/containers/Review/ReviewWrapper.tsx:114]: Sorry, I think I put you wrong. Let's just leave it how it was (hide on mobile) for now, and we can look at this a bit later. I think it needs a bit more before consideration it can work on mobile.

(I'm also thinking that it's generally going to be most useful when there are a lot of stages, so it's less useful for only 1-2 stage applications anyway)

File: src/containers/Review/ReviewWrapper.tsx
Comment (guatemartin) [reply src/containers/Review/ReviewWrapper.tsx]: Oh, I actually did this.  If more than 2 stages, will not show in mobile.

**Review by CarlosNZ — APPROVED** [review]
Cool, working pretty well thanks. There's just a couple of tiny things if you could quickly fix before merging. And I'll make a new issue later on for a few additional tweaks once we've got used to this.

**Comment by guatemartin** [comment]
@CarlosNZ lots of changes here

I just haven't done the "extend date" in overview, as I can't find one that has those properties.

In Overview 
- I removed the left coloured border as without the icon it makes no sense
- reduced the margin of the activities timeline
- removed '.left-area' class that had min-width 300px.  Not needed

Assignment Tab
- Removed the Text Label for the Stage
- Reduced the padding and spacing
- Increased Stage Label size
assignment_tab

Notes Tab
- Left aligned dropdown and toggle button
- Textarea now 'stretches' to use up space
- Reduced spacing
notes_tab_textarea

Documents
- Smaller icon (60 x 60px) could go smaller
- Reduced padding and gap between files
image

**Pull request merged**
